uwblueprint/paramedics-react
# Clean up landing page, and add archive, edit, and delete event functionality

## Overview
>[Code review doc for reference](https://www.notion.so/uwblueprintexecs/Code-Review-a21fc85b00394f488e92d9d605f6b2bc)

[Notion task](https://www.notion.so/uwblueprintexecs/396370c5e09f4c28bc7f455a7e84c908?v=3751aa62a2594c74b7e0ecb363304c24&p=f6dbda0cc5e842eb982ea41c6f6a94dd)


## Changes
- Created common dialog component 
- Added archive event functionality
- Added edit event functionality
- Added unarchive event functionality
- Added delete event functionality
- Added route for editing events, `/events/edit/:eventId`
- Added options menu for `UserProfile` in the `HomeLandingPage`
- Added redirect to `/manage/members` for the `Resource Management` option 

## Testing
- Make sure archiving events is working, and sets the `isActive` field in the db
- Make sure the `Current Events` and `Archived Events` tab sort the events correctly
- Make sure unarchiving works
- Make sure deleting events also deletes events in your db
- Try editing an event, and see if the changes are reflected on the db level
- Try redirecting to the `Resource Management` page

## Screenshots
Archiving Events Dialog
![image](https://user-images.githubusercontent.com/28452372/90317207-b566fa00-def5-11ea-8a26-c5e3922df4dc.png)

Archived Events Moved to Archived Events Tab
![image](https://user-images.githubusercontent.com/28452372/90317213-bf88f880-def5-11ea-9cf4-7f0ed0daec1b.png)

Edit Event Page
![image](https://user-images.githubusercontent.com/28452372/90317220-cf084180-def5-11ea-94a6-a61ed201a0c4.png)

Unarchive Event Dialog
![image](https://user-images.githubusercontent.com/28452372/90317229-dc253080-def5-11ea-8f45-f6671bdeff1c.png)

Delete Event Dialog
![image](https://user-images.githubusercontent.com/28452372/90317236-e5160200-def5-11ea-882b-cb1d873b295f.png)

User Profile Options
![image](https://user-images.githubusercontent.com/28452372/90317245-f232f100-def5-11ea-8eff-8c902112e212.png)

Clicking Resource Management Redirects to Here
![image](https://user-images.githubusercontent.com/28452372/90317258-ff4fe000-def5-11ea-8f1a-aa42c1749704.png)
